src/routes/Routes.tsx

```diff
@@ -42,6 +42,13 @@ const Routes: React.FC = () => (
       <Route exact={true} path="/about" component={AboutPage} />
       <Route exact={true} path="/login" component={SignIn} />
       <Route exact={true} path="/register" component={Register} />
+      {/* // ? what to do here? // */}
+      {/* <Route exact={true} path="/password" component={Password} /> */}
```

Comment:
I'm not quite sure why we need to have a single Password component to deal with both PasswordChange and PassewordReset,  but could have two different routes (one protected and the other one unprotected) using the same componet. So this one could be `/password/reset` and the protected one `password/change`. Or am I missing something.

Reply:
I was thinking of the user who can't change his password from the inside but who can change it using `I forgot my password`, and that lead me to the point where a single Password component can do both jobs _(the reset)_ and _(the change)_, also same for the backend and from that I aimed to have as less code as possible,And as DRY code as possible.

do you think I have to make an addition route as `/reset`? And what do you think about my Idea, I mean for the future is that a good thing?

Reply:
After going through your code I had no difficulty with using the same component(s) for password change and reset in the frontend, but I found it better to separate out these processes in the backend. See my PRs on this branch.